NSSServices/Controllers/VariablesController.cs

```diff
@@ -64,14 +69,136 @@ public override async Task<IActionResult> Get(int id)
             try
             {
                 if(id<0) return new BadRequestResult(); // This returns HTTP 404
-                return Ok(await agent.GetVariable(id));
+                return Ok(agent.GetVariableWithUnits(id));
             }
             catch (Exception ex)
             {
                 return await HandleExceptionAsync(ex);
             }
         }
-        
+
+        [HttpPost(Name = "Add Variable")]
+        [Authorize(Policy = Policy.AdminOnly)]
+        [APIDescription(type = DescriptionType.e_link, Description = "/Docs/Variables/Add.md")]
+        public async Task<IActionResult> Post([FromBody] VariableWithUnit entity)
+        {
+            try
+            {
+                if (!isValid(entity)) return new BadRequestResult(); // This returns HTTP 404
+
+                Variable tempV = new Variable
```

Comment:
Thinking about readability here.. shortening the variables can be somewhat confusing if we're trying to trace things backwards. Generally, try naming things something that is clear as to what it means or consistent with the rest of the code, like `newVariable` since you're creating a new variable or if you want to go with temp, try spelling out `tempVariable`, same for `tempVT`.. `newVariableType`? Since there are a lot of things called `temp` going on here, it's easy to misunderstand which I'm looking at.

Reply:
Variables have been renamed.